lib/automation/report_context_generator.rb

```diff
@@ -41,6 +43,20 @@ def initialize(
             .where("topics.archetype = ?", Archetype.default)
         @posts = @posts.where("categories.read_restricted = ?", false) if !@allow_secure_categories
         @posts = @posts.where("categories.id IN (?)", @category_ids) if @category_ids.present?
+        @posts =
+          @posts.where(
+            "categories.id NOT IN (?)",
+            exclude_category_ids,
+          ) if exclude_category_ids.present?
+
+        if exclude_tags.present?
+          exclude_tag_ids = Tag.where(name: exclude_tags).select(:id)
```

Comment:
Minor, but I think the `where_name` scope should probably be used for this:

https://github.com/discourse/discourse/blob/74edd165a87b861591d40d62d21be5f7b635392c/app/models/tag.rb#L24-L28

Reply:
oh yeah it can work, will change  ... fwiw we have an index on lower(name) ... so yay